app/models/request_token.rb

```diff
@@ -1,18 +1,17 @@
 class RequestToken < OauthToken
-
   attr_accessor :provided_oauth_verifier
 
   def authorize!(user)
     return false if authorized?
     self.user = user
     self.authorized_at = Time.zone.now
-    self.verifier = OAuth::Helper.generate_key(20)[0, 20] unless oauth10?
+    self.verifier = OAuth::Helper.generate_key(20)[0, 20]
     self.save!
   end
 
   def exchange!
     return false unless authorized?
-    return false unless oauth10? || verifier==provided_oauth_verifier
+    return verifier==provided_oauth_verifier
```

Comment:
```
return false unless verifier==provided_oauth_verifier
```

otherwise this returns unconditionally and doesn't execute the transaction below